### PR TITLE
RavenDB-13487_v4.1: Migrating from v3.5 RavenFS throw NRE

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/Smuggler/Migration/Migrator_V3.cs [hunk 1]
```diff
@@ -167,10 +173,10 @@ private async Task<string> MigrateRavenFs(string lastEtag)
                         WriteDocumentWithAttachment(documentActions, context, dataStream, key, metadata);
 
                         Parameters.Result.Documents.ReadCount++;
-                        if (Parameters.Result.Documents.Attachments.ReadCount % 50 == 0 || sp.ElapsedMilliseconds > 3000)
+                        if (Parameters.Result.Documents.ReadCount % 50 == 0 || sp.ElapsedMilliseconds > 3000)
```
Comment (grisha-kotler) [inline src/Raven.Server/Smuggler/Migration/Migrator_V3.cs:176]: The attachment count needs to incremented

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Smuggler/Migration/Migrator_V3.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Smuggler/Migration/Migrator_V3.cs:176]: :+1:

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Smuggler/Migration/Migrator_V2.cs [hunk 2]
```diff
@@ -108,7 +115,7 @@ private async Task MigrateAttachments(string lastEtag)
                     var attachmentsArray = await GetAttachmentsList(lastEtag, transactionOperationContext);
                     if (attachmentsArray.Length == 0)
                     {
-                        var count = Parameters.Result.Documents.Attachments.ReadCount;
+                        var count = Parameters.Result.Documents.ReadCount;
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Migration/Migrator_V2.cs:118]: This looks bizarre? Why are we not using the Attachments here?

Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Migration/Migrator_V2.cs:118]: It is the same number. Migrating legacy attachment creates a new document + attachment.

Comment (garayx) [reply src/Raven.Server/Smuggler/Migration/Migrator_V2.cs:118]: we write the attachment as dummy doc therefore its proper to increase docs count.
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/Smuggler/Migration/AbstractLegacyMigrator.cs#L104

Comment (ppekrol) [reply src/Raven.Server/Smuggler/Migration/Migrator_V2.cs:118]: How this is displayed in the Studio? Shouldn't we increase both numbers?

Comment (garayx) [reply src/Raven.Server/Smuggler/Migration/Migrator_V2.cs:118]: we increase the docs count here and the attachments count increased by DatabaseItemType.Attachments so in studio we see both docs and attachments
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs#L171

**Pull request merged**